core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -995,6 +988,39 @@ private[spark] class AppStatusListener(
     }
   }
 
+  private def updateBroadcastBlock(
+      event: SparkListenerBlockUpdated,
+      broadcast: BroadcastBlockId): Unit = {
+    val now = System.nanoTime()
+    val executorId = event.blockUpdatedInfo.blockManagerId.executorId
+    val storageLevel = event.blockUpdatedInfo.storageLevel
+
+    // Whether values are being added to or removed from the existing accounting.
+    val diskDelta = event.blockUpdatedInfo.diskSize * (if (storageLevel.useDisk) 1 else -1)
+    val memoryDelta = event.blockUpdatedInfo.memSize * (if (storageLevel.useMemory) 1 else -1)
+
+    liveExecutors.get(executorId).foreach { exec =>
```

Comment:
All the code above can be moved within the `foreach`.

Reply:
@vanzin Yes! It is true. I have updated the code.